### ISSUE TITLE
[Bug] [RayService] Missing readiness and liveness probe causes 5xx error

### ISSUE THREAD

**Issue by simon-mo** [ray-project/kuberay#667]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.

### KubeRay Component

Others

### What happened + What you expected to happen

When the Service Operator update a new Ray Cluster. It should install necessary [livenessProbe and readinessProbe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request) targeting specifically to Ray Serve.
- Check the HTTP Proxy has been properly started on each node (by calling localhost:8000/-/healthz) as part of readiness probe.
- Check the application is ready to be served (as part of readiness probe) (by calling Serve's status API, waiting to application to be running)
- Check the application is healthy via status API (as part of liveness probe). 

Two more considerations:
- Should we also health check Ray and GCS health as part of this. As context, the Ray cluster operator will fill in the following probes if there is non specified:

https://github.com/ray-project/kuberay/blob/5e862beebbd5a85c9a84eb44db2bdac5b5b48723/ray-operator/controllers/ray/common/pod.go#L226-L270

- Should we add some helper method in the CLI for this?

The impact of this fix will help reduce the 500s error during startup because it will then make sure only to route traffic to pod when it is ready to accept traffic; rather than suffering from several layers latency of Python startup, Ray startup, and actor startup. 

### Reproduction script

...

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by akanso** [comment]
is there any specific reason why we use _Exec_ with _Wget_ instead of the _HTTPGet_ in the probe?

```go
type ProbeHandler struct {
	// Exec specifies the action to take.
	// +optional
	Exec *[ExecAction](https://pkg.go.dev/k8s.io/kubernetes/pkg/apis/core#ExecAction)
	// HTTPGet specifies the http request to perform.
	// +optional
	HTTPGet *[HTTPGetAction](https://pkg.go.dev/k8s.io/kubernetes/pkg/apis/core#HTTPGetAction)
	// TCPSocket specifies an action involving a TCP port.
	// +optional
	TCPSocket *[TCPSocketAction](https://pkg.go.dev/k8s.io/kubernetes/pkg/apis/core#TCPSocketAction)

	// GRPC specifies an action involving a GRPC port.
	// This is a beta field and requires enabling GRPCContainerProbe feature gate.
	// +featureGate=GRPCContainerProbe
	// +optional
	GRPC *[GRPCAction](https://pkg.go.dev/k8s.io/kubernetes/pkg/apis/core#GRPCAction)
}

```


https://pkg.go.dev/k8s.io/kubernetes/pkg/apis/core#ProbeHandler

**Comment by simon-mo** [comment]
I think the reason might be httpGet doesn't cover multiple endpoints.

**Comment by shrekris-anyscale** [comment]
Yeah, that should be feasible. I can take it.

**Comment by kevin85421** [comment]
@shrekris-anyscale Is this issue resolved? Thanks!

**Comment by shrekris-anyscale** [comment]
Hi @kevin85421 the bug that motivated this suggestion is resolved by #730. However, #730 doesn't use liveness and readiness probes, which design-wise would be a better solution. There's no straightforward ways to check for `RayService` health in the probes at the moment, but I think we should leave this ticket open to track the issue. I've reclassified this from a P1 to a P2.
